boat-maven-plugin/src/main/java/com/backbase/oss/boat/RadioMojo.java

```diff
@@ -0,0 +1,171 @@
+
+        BasicAuthRequestInterceptor basicAuthRequestInterceptor = null;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        if(username != null && !username.isEmpty() && password != null && !password.isEmpty()){
+             basicAuthRequestInterceptor = new BasicAuthRequestInterceptor(username, password);
+        }else{
+            getLog().warn("No Authentication set");
+        }
+
+        List<UploadSpec> allSpecs = Arrays.stream(specs).map(this::mapToUploadSpec).collect(Collectors.toList());
+
+        ApiClient apiClient = new ApiClient().setBasePath(basePath);
+        if(basicAuthRequestInterceptor != null){
+            apiClient.addAuthorization("Basic Auth", basicAuthRequestInterceptor);
+        }
+
+        BoatMavenPluginApi api = apiClient.buildClient(BoatMavenPluginApi.class);
+
+        UploadRequestBody uploadRequestBody = new UploadRequestBody();
+        uploadRequestBody.setGroupId(groupId);
+        uploadRequestBody.setArtifactId(artifactId);
+        uploadRequestBody.setVersion(version);
+        uploadRequestBody.setSpecs(allSpecs);
+
+        List<BoatLintReport> reports = api.uploadSpec(portalKey,sourceKey,uploadRequestBody);
+
+        try {
+            objectMapper.writerWithDefaultPrettyPrinter().writeValue(new File(getOutput(), "radioOutput.json"),reports);
+        } catch (IOException e) {
+            throw new MojoFailureException("Failed to write output", e);
+        }
+
+    }
+
+    private UploadSpec mapToUploadSpec(SpecConfig spec) {
+
+        OpenAPI openApi = null;
+        String contents = null;
+        File file = spec.getInputSpec();
+        try {
+            contents = IOUtils.toString(spec.getInputSpec().toURI(), Charset.defaultCharset());
+            openApi = OpenAPILoader.parse(contents);
+        } catch ( IOException e) {
+            String msg = "Invalid File Path: " + file.getName();
+            getLog().error(msg);
+            throw new RuntimeException(msg, e);
+        } catch (OpenAPILoaderException e) {
+            String msg = "Invalid Open Api file: " + file.getName();
+            getLog().error(msg);
+            throw new RuntimeException(msg, e);
+        }
+
+        String key = spec.getKey();
+        if (key == null || key.isEmpty()) {
+            key = file.getName().substring(0, file.getName().lastIndexOf("-"));
+        }
+
+        String name = spec.getName();
+        if (name == null || name.isEmpty()) {
+            name = file.getName();
+        }
+
+        UploadSpec uploadSpec = new UploadSpec();
+        uploadSpec.setFileName(file.getName());
+        uploadSpec.setKey(key);
+        uploadSpec.setName(name);
+        uploadSpec.setOpenApi(contents);
+
+
+        return uploadSpec;
+
+    }
+
+    @SneakyThrows
+    private File getOutput() {
+        if (radioOutput == null) {
+            radioOutput = new File("./target/boat-radio-report");
+        }
+        if(!radioOutput.exists()){
+            radioOutput.mkdir();
```

Comment:
Do we want to use `mkdirs` which will also make any missing parent directories?

Reply:
Done